drivers/include/hmc5883l.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_hmc5883l HMC5883L 3-axis digital compass
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Honeywell HMC5883L 3-axis digital compass
+ *
+ * The driver implements basic polling mode. The application can use
+ * different approaches to get new data, either
+ *
+ * - using the #hmc5883l_read function at a lower rate than the the DOR, or
+ * - using the data-ready interrupt (**DRDY**), see #hmc5883l_init_int.
+ *
+ * The data-ready interrupt (**DRDY**) is only be available when module
+ * `hmc5883l_int` is enabled.
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ *
+ * @{
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#ifndef HMC5883L_H
+#define HMC5883L_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+/** HMC5883L I2C address*/
+#define HMC5883L_I2C_ADDRESS    (0x1e)
+
+/** HMC5883L chip id defined in Identification Registers A..C */
+#define HMC5883L_ID             { 0x48, 0x34, 0x33 }
+
+/** HMC5883L chip id length */
+#define HMC5883L_ID_LEN         (3)
+
+/** Definition of error codes */
+typedef enum {
+    HMC5883L_OK,                   /**< success */
+    HMC5883L_ERROR_I2C,            /**< any I2C communication error */
+    HMC5883L_ERROR_WRONG_ID,       /**< wrong id read */
+    HMC5883L_ERROR_NO_DATA,        /**< no data are available */
+    HMC5883L_ERROR_RAW_DATA,       /**< reading raw data failed */
+    HMC5883L_ERROR_COMMON,         /**< common error */
+} hmc5883l_error_codes_t;
+
+/**
+ * @brief   Data output rates (DOR)
+ *
+ * Values correspond to bits <4:2> of HMC5883L_REG_CFG_A register
+ */
+typedef enum {
+
```

Comment:
I am not a fan of these newlines here and below, and they are not common in the RIOT code base. In any case, the formatting should at least be consistent inside this file :-)

Reply:
It's probably a matter of taste. According to style guide this is not forbidden. Personally, I find it easier to read. But, I can change it.